ocs_ci/deployment/fusion_aas.py

```diff
@@ -0,0 +1,157 @@
+                rosa_prod_cluster.create_admin_and_login()
+                rosa_prod_cluster.generate_kubeconfig_file(skip_tls_verify=True)
+                rosa_prod_cluster.generate_kubeadmin_password_file()
+
+        self.test_cluster()
+
+
+class FUSIONAAS(rosa_deployment.ROSA):
+    """
+    Deployment class for Fusion aaS.
+    """
+
+    OCPDeployment = FUSIONAASOCP
+
+    def __init__(self):
+        self.name = self.__class__.__name__
+        super(FUSIONAAS, self).__init__()
+        ocm.download_ocm_cli()
+        rosa.download_rosa_cli()
+        self.aws = AWSUtil(self.region)
+
+    def deploy_ocp(self, log_cli_level="DEBUG"):
+        """
+        Deployment specific to OCP cluster on a cloud platform.
+
+        Args:
+            log_cli_level (str): openshift installer's log level
+                (default: "DEBUG")
+
+        """
+        ocm.login()
+        super(FUSIONAAS, self).deploy_ocp(log_cli_level)
+        self.host_network_update()
+
+    def deploy_ocs(self):
+        """
+        Deployment of ODF Managed Service addon on Fusion aaS.
+        """
+        ceph_cluster = ocp.OCP(kind="CephCluster", namespace=self.namespace)
+        try:
+            ceph_cluster.get().get("items")[0]
+            logger.warning("OCS cluster already exists")
+            return
+        except (IndexError, CommandFailed):
+            logger.info("Running OCS basic installation")
+        create_fusion_monitoring_resources()
+        deploy_odf()
+        pod = ocp.OCP(kind=constants.POD, namespace=self.namespace)
+
+        # Check for Ceph pods
+        assert pod.wait_for_resource(
+            condition="Running",
+            selector=constants.MON_APP_LABEL,
+            resource_count=3,
+            timeout=1200,
+        )
+        assert pod.wait_for_resource(
+            condition="Running", selector=constants.MGR_APP_LABEL, timeout=600
+        )
+        assert pod.wait_for_resource(
+            condition="Running",
+            selector=constants.OSD_APP_LABEL,
+            resource_count=3,
+            timeout=600,
+        )
+
+        # Verify health of ceph cluster
+        ceph_health_check(namespace=self.namespace, tries=60, delay=10)
+
+    def destroy_ocs(self):
+        """
+        Uninstall ODF Managed Service addon via rosa cli.
+        """
+        cluster_namespace = config.ENV_DATA["cluster_namespace"]
+
+        # Deleting PVCs
+        rbd_pvcs = [
+            p
+            for p in pvc.get_all_pvcs_in_storageclass(constants.CEPHBLOCKPOOL_SC)
+            if not (
+                p.data["metadata"]["namespace"] == cluster_namespace
+                and p.data["metadata"]["labels"]["app"] == "noobaa"
+            )
+        ]
+        pvc.delete_pvcs(rbd_pvcs)
+        cephfs_pvcs = pvc.get_all_pvcs_in_storageclass(constants.CEPHFILESYSTEM_SC)
+        pvc.delete_pvcs(cephfs_pvcs)
+        logger.error("OCS NOT IMPLEMENTED")
```

Comment:
Currently, we can't automate the OCS deletion with `FUSIONAAS`?

Reply:
We can automate deletion of namespaces but it sometimes causes issues with leftovers and finalizers. The decision to automate it could be determined on one of our mtgs. I would provide it in a different PR when we have consensus on the approach.

Reply:
I see, thanks.